src/wasm/wasm.cpp

```diff
@@ -1318,6 +1319,71 @@ void StringSliceIter::finalize() {
   }
 }
 
+ArenaVector<Type>& Resume::getSentTypes() {
+  if (this->sentTypes.size() != this->handlerBlocks.size()) {
+    Fatal() << "Types sent to blocks have not been determined, yet.";
+  }
+  return this->sentTypes;
+}
+
+void Resume::finalize() {
+  // Only performing some validation, but no finalization. The finalize(Module*)
+  // function must have been called previously.
+
+  if (!(this->contType.isContinuation() &&
+        this->contType.getContinuation().type.isSignature())) {
+    Fatal() << "ill-formed Resume expression";
+  }
+
+  if (this->sentTypes.size() != this->handlerBlocks.size()) {
+    Fatal() << "Resume node was not finalized";
+  }
+}
+
+void Resume::finalize(Module* wasm) {
+  if (!(this->contType.isContinuation() &&
+        this->contType.getContinuation().type.isSignature())) {
+    Fatal() << "ill-formed Resume expression";
+  }
```

Comment:
Can we move this (validation in `Resume::finalize()` and `Resume::finalize(Module* wasm)` to `wasm-validator.cpp`? Also if we do
```cpp
void finalize(Module* wasm = nullptr);
```
in `wasm.h`, we wouldn't need two methods.

Reply:
Done